pkg/kube/util.go

```diff
@@ -91,21 +90,10 @@ func BuildClientCmd(kubeconfig, context string, overrides ...func(*clientcmd.Con
 	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
 }
 
-// CreateClientset is a helper function that builds a kubernetes Clienset from a kubeconfig
-// filepath. See `BuildClientConfig` for kubeconfig loading rules.
-func CreateClientset(kubeconfig, context string, fns ...func(*rest.Config)) (*kubernetes.Clientset, error) {
-	c, err := BuildClientConfig(kubeconfig, context)
-	if err != nil {
-		return nil, fmt.Errorf("build client config: %v", err)
-	}
-	for _, fn := range fns {
-		fn(c)
-	}
-	return kubernetes.NewForConfig(c)
-}
-
-// NewRestConfigFromContext returns the rest.Config for the given kube config context.
-func NewRestConfigFromContext(kubeConfig []byte, configOverrides ...func(*rest.Config)) (*rest.Config, error) {
+// NewRemoteRestConfig returns the rest.Config for the given kube config context.
+// This is suitable for access to remote clusters from untrusted kubeConfig inputs.
+// The kubeconfig is sanitized and unsafe auth methods are denied.
+func NewRemoteRestConfig(kubeConfig []byte, configOverrides ...func(*rest.Config)) (*rest.Config, error) {
```

Comment:
I donot see the need to call `Remote`, it can also be used for local cluster if kubeconfig bytes provided

Reply:
Renamed to NewUntrustedRestConfig. Its specifically about trusted vs untrusted, not remove vs local, so now the name reflects this.